### ISSUE TITLE
Updating scrollablePlotArea.minHeight breaks yAxis and bar category labels

### ISSUE THREAD

**Issue by mayhewluke** [highcharts/highcharts#14277]
#### Expected behaviour

* Updating `scrollablePlotArea.minHeight` redraws the chart so that the y-axis remains at the proper location.
* Updating `scrollablePlotArea.minHeight` redraws the chart so that the bar categories labels are redrawn if there is now space for them/no longer space for them.

#### Actual behaviour

1. If `yAxis` is present, changing `minHeight` doesn't update `yAxis` position properly (it gets "stuck" in the middle of the chart, if the chart has beccome bigger).
2. If `series`/`categories` are increased and *then* `minHeight` is increased, canvas size increased but bars aren't redrawn, so it just creates extra blank space.
3. If `series`/`categories` and `minHeight` are updated simultaneously, the category labels aren't updated (some will be missing when increasing the chart size). Only if `series`/`categories` is set *after* `minHeight` does it update the labels.

#### Live demo with steps to reproduce

https://codesandbox.io/s/beautiful-hellman-7tnvp

![highcharts-minheight-bugs](https://user-images.githubusercontent.com/5531936/94448109-c5fec700-0167-11eb-9f2a-6e4e4882cd26.gif)

Note: reload the page in between reproduction attempts to reset the chart state.

**Bug #​1 reproduction steps:**

1. Click `Toggle y-axis definition`.
2. Click `Update entire options object`  (y-axis label should update to `My y-axis`).
3. Click `Toggle size`.
4. Click `Update entire options object` (increases the bars/minHeight).
5. Click `Toggle size`.
6. Click `Update entire options object` (decreases the bars/minHeight).
7. Click `Toggle size`.
8. Click `Update entire options object` (increases the bars/minHeight again).
9. Observe that the y-axis is in the wrong location.
10. Reload and do the same with the yAxis definition turned *off*, and the yAxis will behave as expected.

**Bug #​2 reproduction steps:**

1. Click `Toggle size`.
2. Click `Update series/categories`.
3. Click `Update minHeight`.
4. Observe that the canvas size has increased, but the bars haven't been redrawn to fill the space.

**Bug #​3 reproduction steps:**

1. Click `Toggle size`.
2. Click `Update entire options object` or `Update minHeight and series/categories simultaneously`.
3. Observe that some bar categories don't have labels even though there is now space for them.
4. Click `Update series/categories`.
5. Observe that the bars categories are now redrawn so that all have labels.

#### Product version

Highcharts v8.2.0

#### Affected browser(s)

Chrome and Firefox, haven't tried any others.

**Comment by PaulDalek** [comment]
Hi @mayhewluke

Thank you for reporting these issues.

**Issue 1**
This behavior has been already reported (#12112) and as a workaround you can call the `moveFixedElements` function.

**Issue 2 and 3**
It looks like the series isn’t redrawn correctly after the `scrollablePlotArea` is updated. In both cases you can force to redraw the `series` again by setting the `isDirty` flag to `true` and calling the `chart.redraw`.

**Demo**
https://jsfiddle.net/BlackLabel/0huqmsnb/

**Comment by mayhewluke** [comment]
Hi @PaulDalek, thanks for the feedback and the workarounds. Unfortunately we're using the React wrapper, and I couldn't get the workarounds to work properly in React. They did help me figure out the following solutions/workarounds though:

* For issue 3, a workaround is to set `{ xAxis: { labels: { step: 1 } } }`.
* For issue 1, your tip about `moveFixedElements` helped me find a way to actually fix the issue in the Highcharts source code, so I've opened a PR for that (#14284).
* Issue 2 doesn't actually affect us, it's just something I found in exploring the other two issues, so I haven't bothered with a fix/workaround for it at this point.
